src/components/FloatingMenu/FloatingMenu.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2023 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import classnames from "classnames";
+import React, { ComponentPropsWithoutRef, ReactNode, forwardRef } from "react";
+import styles from "./FloatingMenu.module.css";
+import useId from "../../utils/useId";
+import { Text } from "../Typography/Text";
+
+interface Props extends ComponentPropsWithoutRef<"div"> {
+  /**
+   * The size of the menu in px.
+   */
+  // TODO: Figure out whether this component should come with a standard set of
+  // T-shirt sizes
+  size: number;
+  /**
+   * The menu title.
+   */
+  title: string;
+  /**
+   * The CSS class.
+   */
+  className?: string;
+  /**
+   * The menu contents.
+   */
+  children: ReactNode;
+}
+
+/**
+ * A floating menu (the kind of menu you would see on desktop).
+ */
+// This an internal component not intended for export! Consumers should use it
+// via the Menu or ContextMenu components (which at the time of writing, are not
+// yet built).
```

Comment:
What's the difference between `Menu` and `ContextMenu`?

If this is an internal, could we instead bury this within the `ContextMenu` folder and not export it? So it can't be mistakenly imported and incorrectly used

Reply:
For the first point: I've now expanded the comment

For the second point: I've renamed the folder to Menu to reflect the fact that it'll be buried as an implementation detail of the Menu component. Not sure what you mean about not exporting it though, because we need to export the component for Storybook usage and tests - as long as we don't export it from index.ts, that should be enough to make sure it stays internal, I think?